ravedude/src/avrdude/mod.rs

```diff
@@ -17,6 +17,35 @@ pub struct Avrdude {
 }
 
 impl Avrdude {
+    fn get_avrdude_version() -> anyhow::Result<(u8, u8)> {
+        let output = process::Command::new("avrdude").arg("-?").output()?;
+        let stderr: &str = std::str::from_utf8(&output.stderr)?;
+        let err = || anyhow::anyhow!("failed to derive version number from avrdude");
+        let version: &str = stderr.split("version").last().ok_or_else(err)?.trim();
+        let version: String = version
+            .chars()
+            .take_while(|c| c.is_ascii_digit() || *c == '.')
+            .collect();
+        let (major, minor) = version.split_once('.').ok_or_else(err)?;
+        let major = major.parse::<u8>()?;
+        let minor = minor.parse::<u8>()?;
+        Ok((major, minor))
+    }
+
+    pub fn require_min_ver((req_major, req_minor): (u8, u8)) -> anyhow::Result<()> {
+        let (major, minor) =
+            Self::get_avrdude_version().context("Failed reading avrdude version information.")?;
+        if (major, minor) < (req_major, req_minor) {
+            anyhow::bail!(
+                "Avrdude does not meet minimum version requirements. v{}.{} was found while v{}.{} or greater is required.\n\
+                You may find a more suitable version here: http://download.savannah.gnu.org/releases/avrdude/",
```

Comment:
One minor nit, please make this an `https` link so people don't go through an unsecured connection if their browser isn't strict about encryption.

```suggestion
                You may find a more suitable version here: https://download.savannah.gnu.org/releases/avrdude/",
```